src/transformations.jl

```diff
@@ -62,6 +62,11 @@ struct ZScoreTransform{T<:Real} <: AbstractDataTransform
         lens == l || lens == 0 || throw(DimensionMismatch("Inconsistent dimensions."))
         new{T}(l, dims, m, s)
     end
+
+    function ZScoreTransform(dims::Int, m::Vector{T}, s::Vector{T}) where T
+        Base.depwarn("ZScoreTransform(dims, m, s) is deprecated: use ZScoreTransform(len, dims, m, s) instead", :ZScoreTransform)
+        ZScoreTransform(max(length(m), length(s)), dims, m, s)
+    end
```

Comment:
I think you got the deprecation backwards: it's `dims` that was introduced by https://github.com/JuliaStats/StatsBase.jl/pull/490, not `len`:
```suggestion
    function ZScoreTransform(l::Int, m::Vector{T}, s::Vector{T}) where T
        Base.depwarn("ZScoreTransform(len, m, s) is deprecated: use ZScoreTransform(len, 2, m, s) instead", :ZScoreTransform)
        ZScoreTransform(l, 2, m, s)
    end
```
(BTW, this doesn't have to be an inner constructor, though it doesn't really make a difference I guess.)

Reply:
Won't this cause problems if you try to apply this transformation to a vector though?

Reply:
Although I guess that given this is not something users are likely to directly interact with very often then that's okay.

Reply:
Well it should work the same as it did in 0.31 I guess: we just deprecated the implicit `dims`, but the behavior with `dims=2` isn't supposed to have changed IIRC.